services/wallet/router/pathprocessor/processor_swap_paraswap.go

```diff
@@ -172,7 +173,7 @@ func (s *SwapParaswapProcessor) BuildTx(params ProcessorInputParams) (*ethTypes.
 func (s *SwapParaswapProcessor) prepareTransaction(sendArgs *MultipathProcessorTxArgs) error {
 	slippageBP := uint(sendArgs.SwapTx.SlippagePercentage * 100) // convert to basis points
 
-	key := makeKey(sendArgs.SwapTx.ChainID, sendArgs.SwapTx.ChainIDTo, sendArgs.SwapTx.TokenIDFrom, sendArgs.SwapTx.TokenIDTo)
+	key := makeKey(sendArgs.SwapTx.ChainID, sendArgs.SwapTx.ChainID, sendArgs.SwapTx.FromSymbol, sendArgs.SwapTx.ToSymbol)
```

Comment:
Why don't we let it as it was?

Reply:
I can do if you feel that's better, I just didn't think we were using the proper names for what we wanted to store in those fields.

Reply:
I was mostly referring to `sendArgs.SwapTx.ChainID, sendArgs.SwapTx.ChainIDTo` part, so would like to keep it as it was `sendArgs.SwapTx.ChainID, sendArgs.SwapTx.ChainIDTo`, because of `AvailableFor` function the router logic will never allow calling `prepareTransaction` (via `BuildTransaction` or `Send`) with different chains.

For the change `TokenIDFrom` -> `FromSymbol` as you wish, cause as said in the previous comment we plan to optimize that part (and yes as you pointed already we can add then props tokenID to be used for collectibles and Symbol from/to to be used for other sending types).